GPflowOpt/optim.py

```diff
@@ -240,18 +255,21 @@ def optimize(self, objectivefx):
         """
 
         self.optimizers[0].set_initial(self.get_initial())
-        fun_evals = []
+        results = []
         for current, next in zip(self.optimizers[:-1], self.optimizers[1:]):
             result = current.optimize(objectivefx)
-            fun_evals.append(result.nfev)
+            results.append(result)
             if not result.success:
                 result.message += " StagedOptimizer interrupted after {0}.".format(current.__class__.__name__)
                 break
-            next.set_initial(result.x)
```

Comment:
It's probably a good idea not to use reserved keywords to avoid confusion?

forward.set_initial(...)
following.set_initial(...)